fetch-in-react/my-app/src/UserDetails.jsx

```diff
@@ -0,0 +1,48 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { useEffect, useState } from 'react';
+import UserCard from './UserCard';
+
+export default function UserDetails({ userId, onCancel }) {
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState();
+  const [user, setUser] = useState();
+
+  /* your code here (hint: useEffect) */
+
+  useEffect(() => {
+    async function getUserInfo() {
+      try {
+        const response = await fetch(
+          `https://jsonplaceholder.typicode.com/users/${userId}`
+        );
+        if (!response.ok) {
+          setError(`HTTP error! Status: ${response.status}`);
```

Comment:
Since you don't return here (or use an `else` clause), after setting the error you will continue to get the JSON response and set the user. There is a chance that will throw an exception.

The easiest way to handle this is to `throw new Error(message);` here and let the `catch` clause call `setError`. That way you are guaranteed to _not_ execute lines 21 and 22.